frame/dapps-staking/src/tests.rs

```diff
@@ -1264,6 +1264,50 @@ fn new_era_forcing() {
     })
 }
 
+#[test]
+fn new_era_length_is_always_blocks_per_era() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+        let blocks_per_era = mock::BLOCKS_PER_ERA;
+
+        // start new era era and record era number and height
+        advance_to_era(mock::DappsStaking::current_era() + 1);
+        let mut expected_era = mock::DappsStaking::current_era();
+        let mut previous_era_starting_block_height;
+        let mut current_era_starting_block_height = System::block_number();
+
+        // show normal era advancement works and its length is blocks_per_era
+        advance_to_era(mock::DappsStaking::current_era() + 1);
+        expected_era += 1;
+        previous_era_starting_block_height = current_era_starting_block_height;
+        current_era_starting_block_height = System::block_number();
+        assert_eq!(mock::DappsStaking::current_era(), expected_era);
+        assert_eq!(current_era_starting_block_height - previous_era_starting_block_height, blocks_per_era);
+
+
+        // now, move to middle of an era, force new era and assert the next block starts a new era
+        run_for_blocks(1);
```

Comment:
Perhaps make this dependent on the blocks per era constant?

E.g. if BLOCKS_PER_ERA gets modified in the future, you'd want your UT to still work.

Reply:
I would have usually done some value related to division of blocks_per_era, but I decided to hardcode this since blocks_per_era for the local runtime was only 3 blocks, which is the smallest era value to realistically test this (although technically 2 could work). I don't expect this to change Down to 2, and the test won't break as long as blocks_per_era >= 3.